core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorBodyDirective.js

```diff
@@ -38,10 +42,39 @@ oppia.directive('simpleEditorBody', [function() {
         $scope.canAddNewQuestion = SimpleEditorManagerService.canAddNewQuestion;
         $scope.addState = SimpleEditorManagerService.addState;
         $scope.addNewQuestion = SimpleEditorManagerService.addNewQuestion;
-        $scope.canTryToPublish = SimpleEditorManagerService.canTryToPublish;
+        $scope.canTryToFinishExploration =
+          SimpleEditorManagerService.canTryToFinishExploration;
 
-        $scope.tryToPublishExploration = function() {
-          // TODO(sll): Implement this.
+        $scope.isExplorationFinishable = function() {
+          if (explorationRightsService.isPrivate()) {
+            if (!explorationWarningsService.countWarnings()) {
+              return true;
+            }
+          } else if (explorationSaveService.isExplorationSaveable()) {
+            return true;
+          }
+
+          return false;
+        };
+
+        $scope.startFinishExploration = function() {
```

Comment:
Let's just call this finishExploration -- startFinishExploration is a bit of an oxymoron, and we don't really want to start introducing things like finishFinishExploration :P

That said, what do you think about publishChanges as an alternative name?